cryptography/src/commonJvmAndroid/kotlin/com.wire.kalium.cryptography/MLSClientImpl.kt

```diff
@@ -144,28 +144,34 @@ class MLSClientImpl(
     @Suppress("TooGenericExceptionCaught")
     override suspend fun decryptMessage(groupId: MLSGroupId, message: ApplicationMessage): List<DecryptedMessageBundle> {
         var decryptedMessage: DecryptedMessage? = null
+        var capturedError: Throwable? = null
 
         coreCrypto.transaction(object : CoreCryptoCommand {
             override suspend fun execute(context: CoreCryptoContext) {
                 try {
-                    val result = context.decryptMessage(
+                    decryptedMessage = context.decryptMessage(
                         groupId.decodeBase64Bytes(),
                         message
                     )
-                    decryptedMessage = result
-
                 } catch (throwable: Throwable) {
                     val isBufferedFutureError = (
                             throwable is CoreCryptoException.Mls && throwable.v1 is MlsException.BufferedFutureMessage
-                            ) || throwable.message
-                        ?.contains("Incoming message is a commit for which we have not yet received all the proposals") == true
-                    if (!isBufferedFutureError) {
-                        throw throwable
+                            ) || throwable.message?.contains(
+                        "Incoming message is a commit for which we have not yet received all the proposals"
+                    ) == true
+
+                    if (isBufferedFutureError) {
+                        return
                     }
+
+                    capturedError = throwable
+                    throw throwable
                 }
             }
         })
 
+        capturedError?.let { throw it }
```

Comment:
I think you need a try catch around `coreCrypto.transaction` otherwise you'll never reach this code when an error is thrown.